### PR TITLE
Improve Kiteshield rule

### PR DESCRIPTION
**Pull request by egibs**
This PR is a follow-up to https://github.com/chainguard-dev/bincapz/pull/240 and implements both suggestions:
1. Removes `import "elf"`
2. Supports `aarch64` in addition to `x86_64`

I did a bit of `xxd` and Endianness work to help populate these, e.g.:
`x86_64` (`0x003e`):
```
1cedc8fb2e76:/# xxd -e -l 64 /bin/sh
00000000: 464c457f 00010102 00000000 00000000   .ELF............
00000010: 003e0003 00000001 0000b5b0 00000000   ..>.............
00000020: 00000040 00000000 00087358 00000000   @.......Xs......
00000030: 00000000 00380040 0040000d 001a001b   ....@.8...@.....
```
`aarch64` (`0x00b7`):
```
8c95055dbbb5:/# xxd -e -l 64 /bin/sh
00000000: 464c457f 00010102 00000000 00000000   .ELF............
00000010: 00b70003 00000001 0000a780 00000000   ................
00000020: 00000040 00000000 000a0348 00000000   @.......H.......
00000030: 00000000 00380040 00400009 00180019   ....@.8...@.....
```
`ET_EXEC` (`0x0002`):
```
8c95055dbbb5:/# xxd -e -l 64 /usr/bin/go
00000000: 464c457f 00010102 00000000 00000000   .ELF............
00000010: 00b70002 00000001 000878b0 00000000   .........x......
00000020: 00000040 00000000 00b8d4e8 00000000   @...............
00000030: 00000000 00380040 00400006 000f0010   ....@.8...@.....
```

### CODE REVIEW

**Comment by egibs** [comment]
[This](https://github.com/chainguard-dev/bincapz/blob/main/rules/evasion/elf-funky-tenable.yara) rule may be another good candidate to improve in a future PR since it also imports `elf`.

File: rules/tools/kiteshield/kiteshield.yara [hunk 1]
```diff
@@ -36,7 +34,19 @@ rule kiteshield : high {
...
+    // EM_X86_64
+    $em_x86_64 = {3e 00}
+    // EM_AARCH64
+    $em_aarch64 = {b7 00}
 
   condition:
-    $loader_jmp and all of ($loader_s*) and elf.type==elf.ET_EXEC and elf.machine == elf.EM_X86_64
+    $loader_jmp 
+    and all of ($loader_s*) 
+    and $elf_magic at 0
```
Comment (egibs) [inline rules/tools/kiteshield/kiteshield.yara:49]: Tested this:
```
$ yara rules/tools/kiteshield/kiteshield.yara /bin/gofmt -c
warning: rule "kiteshield" in rules/tools/kiteshield/kiteshield.yara(40): string "$et_exec" may slow down scanning
warning: rule "kiteshield" in rules/tools/kiteshield/kiteshield.yara(42): string "$em_x86_64" may slow down scanning
warning: rule "kiteshield" in rules/tools/kiteshield/kiteshield.yara(44): string "$em_aarch64" may slow down scanning
1
```

Comment (egibs) [reply rules/tools/kiteshield/kiteshield.yara]: Better implementation: [`fc00476` (#254)](https://github.com/chainguard-dev/bincapz/pull/254/commits/fc0047621303b5150d65c72c322bece5b8ecd3ed)

**Review by tstromberg — APPROVED** [review]
Looks good. Do you mind also moving this rule to `evasion/packer`?

**Comment by egibs** [comment]
> Looks good. Do you mind also moving this rule to `evasion/packer`?

Moved in [`7514889` (#254)](https://github.com/chainguard-dev/bincapz/pull/254/commits/7514889fe59ecc2915e20289817e24d29483c4dd).

**Pull request merged**